packages/rolldown/tests/fixtures/misc/target/es2015/assert.mjs

```diff
@@ -0,0 +1,15 @@
+import { es2015 } from './dist/main'
+import assert from 'node:assert'
+
+assert.strictEqual(es2015.toString(), `function es2015() {
+	let temp;
+	try {
+		temp = JSON.parse("[1, 2, [3]]");
+	} catch (_unused) {}
+	console.log(temp);
+	console.log([
+		1,
+		2,
+		[3]
+	].flat());
```

Comment:
It seems that oxc didn't handle the `flat` in `ES2019`

Reply:
Don't worry. Oxc's Lowering transform is still under development. And it's not rolldown's job to ensure transform output correctness, so this test is not needed.

Reply:
i don't think the lowering transforms typically pollyfil these methods.

e.g. using swc and `promise.withResolvers` (an es2024 feature):

https://play.swc.rs/?version=1.10.3&code=H4sIAAAAAAAAA0vOzysuUahWKCjKz80sTtVRKEotzs8pAzOyUpNLFGoVbBUCIJJ65ZklGUEQ%2BaJiDU0ABRUIQzwAAAA%3D&config=H4sIAAAAAAAAA1VPOw6DMAzdc4rIc4e2Q6X2Dj1ElBoUlJ9iIxUh7k4ICS2b38%2FPnoWUMJCGl5zzmEFUiTAdODM0eVbfzABqp0gnExkuTR1okzplCQu17AqwSj1ySdH9envWBNgQCFuics54003%2FnTq4mJDobNysyvcWz42itoILn7GI9ReeIu4XPOBnamXHYjD0bklOI4plBRJlp3MXAQAA

Reply:
> i don't think the lowering transforms typically pollyfil these methods.

Got it.